server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/BuildComponentTreeStep.java

```diff
@@ -93,14 +91,21 @@ public void execute() {
   }
 
   private ComponentKeyGenerator loadKeyGenerator() {
-    return Stream.of(analysisMetadataHolder.getBranch(), Optional.of(new DefaultKeyGenerator()))
+    return Stream.of(analysisMetadataHolder.getBranch(), Optional.of(new DefaultBranchImpl()))
       // TODO pull request generator will be added here
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135)